internal/security/pkiinit/option/cache.go

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+package option
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// Cache option....
+func Cache() func(*PkiInitOption) (exitCode, error) {
+	return func(pkiInitOpton *PkiInitOption) (exitCode, error) {
+
+		if isCacheNoOp(pkiInitOpton) {
+			return normal, nil
+		}
+
+		return cachePkis(false)
+	}
+}
+
+func isCacheNoOp(pkiInitOption *PkiInitOption) bool {
+	// nop: if the flag is missing or not on
+	return pkiInitOption == nil || !pkiInitOption.CacheOpt
+}
+
+func cachePkis(cacheca bool) (statusCode exitCode, err error) {
+	// generate a new one if pkicache dir is empty
+	pkiCacheDir := getPkiCacheDirEnv()
+	cachedCAPrivateKeyFile := filepath.Join(pkiCacheDir, caServiceName, tlsSecretFileName)
+	if empty, err := isDirEmpty(pkiCacheDir); err != nil {
+		return exitWithError, err
+	} else if empty {
+		// perform generate func
+		statusCode, err = generatePkis()
```

Comment:
Abstracting this capability would allow for the `make test` target to not have to build the `security-secrets-setup` service

Reply:
Good call, I am thinking towards that direction of refactory as well- but not the whole `generatePkis()` because that could greatly reduce the test coverage due to the mocked one.  Maybe abstracting out the calling of real binary action part to remove that `make test` dependency issue as one solution that it can be used in unit tests.

Reply:
@tsconn23 Trevor, I have done the code re-factory regarding the pkisetup build dependency issue of `make test`.  Take a look and move to another issue you mentioned about re-uses some of the existing codes.